src/AutorestSwift/Code Generator/Manager.swift

```diff
@@ -33,56 +33,68 @@ struct StringDecodeKeys: ExtensionStringDecodeKeys {
 
 /// Handles the configuration and orchestrations of the code generation process
 class Manager {
-    internal enum InvocationMode {
-        case autorest
-        case commandLine
-    }
+    static var shared = Manager()
 
     // MARK: Properties
 
-    let inputString: String
-
-    let destinationRootUrl: URL
+    var configured: Bool
+    var config: ManagerConfig
```

Comment:
I wonder if it is a good idea to change config to ManagerConfig? where `nil` means it is not configured and not nil mean it is configured. That way, we don't need to maintain an extra flag `configured` to indicates whether it is configured or not.

Reply:
Great point